src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -42,12 +40,42 @@ class GitVersionPlugin implements Plugin<Project> {
             }
         }
 
+        project.ext.versionDetails = {
+            File gitDir = getRootGitDir(project.rootDir)
+
+            try {
+                Git git = Git.wrap(new FileRepository(gitDir))
+                String description = git.describe().call();
```

Comment:
It'd be good to encapsulate the calls. I've been meaning to clean this up to memoize anyway, can leave as-is and I'll come back to it later.